### PR TITLE
Generated_api_docs

### PR DESCRIPTION
**Pull request by andreasWallnerIFX**
Adds examples to all of the functions a user of the geneated crate will have to use to ease adoption.

### CODE REVIEW

**Comment by andreasWallnerIFX** [comment]
The current formatting fail seems to stem from the source branch, I can rebase to master / wait for "20..." to be clean or we ignore the check when merging to the branch?

File: templates/rust/common.tera [hunk 1]
```diff
@@ -115,31 +115,86 @@ pub struct RegValueT<Reg: sealed::RegSpec> {
...
     fn get_raw(&self) -> T::DataType {
         self.data
     }
-    /// Return a copy with register value set to `value` and write mask fully set
+
+    /// Prepare value to written to register with an arbitrary value
+    ///
+    /// Use this function for setting a register to a custom value, independent
+    /// of bitfields, enumerations, etc. No checks are performed on the passed
+    /// value.
```
Comment (pellico) [inline templates/rust/common.tera:191]: I would add that value is masked with bitfield mask and we guarantee that the adjacent bitfield are not affected.

File: templates/rust/common.tera [hunk 2]
```diff
@@ -115,31 +115,86 @@ pub struct RegValueT<Reg: sealed::RegSpec> {
...
+    /// // needs: use pac::{RegisterValue, TIMER}
+    /// let x = TIMER.bitfield_reg().read().get_raw();
+    /// ```
     #[inline(always)]
     fn get_raw(&self) -> T::DataType {
         self.data
     }
-    /// Return a copy with register value set to `value` and write mask fully set
+
+    /// Prepare value to written to register with an arbitrary value
```
Comment (pellico) [inline templates/rust/common.tera:187]: I would rephrase. "Prepare a register value that could be written to register ...."

File: templates/rust/common.tera [hunk 3]
```diff
@@ -115,31 +115,86 @@ pub struct RegValueT<Reg: sealed::RegSpec> {
...
     #[inline(always)]
     fn new(data: T::DataType) -> RegValueT<T> {
         Self {
             data,
             mask: 0x0u8.into(),
         }
     }
-    /// Get raw register value
+
+    /// Unpack raw integer from value read from register
```
Comment (pellico) [inline templates/rust/common.tera:175]: I would change `Unpack` with `Get`. The term pack sound like the pack layout.

File: templates/rust/common.tera [hunk 4]
```diff
@@ -115,31 +115,86 @@ pub struct RegValueT<Reg: sealed::RegSpec> {
...
+    /// let to_write = timer::BitfieldReg::new(0xdeadbeef);
+    /// TIMER.bitfield_reg().write(to_write);
+    /// let to_write = to_write.boolw().set(true);
+    /// TIMER.bitfield_reg().write(to_write);
+    /// ```
     #[must_use]
     fn new(data: T::DataType) -> Self;
-    /// Get raw register value
+
+    /// Unpack raw integer from value read from register
```
Comment (pellico) [inline templates/rust/common.tera:131]: I would change `Unpack` with `Get`. The term pack sound like the pack layout.

File: templates/rust/common.tera [hunk 5]
```diff
@@ -268,7 +361,7 @@ where
     A: Write,
 RegValueT<T>: Default,
 {
-    /// Init register with value returned by the closure.
+    /// Write register with value built from default register value
```
Comment (pellico) [inline templates/rust/common.tera:364]: I would change to 
`/// Write register with register value built from default register value`

File: templates/rust/common.tera [hunk 6]
```diff
@@ -538,12 +689,20 @@ where
         }
     }
 
+    /// Get mask for bitfield, mask is unshifted and at the actual offset
```
Comment (pellico) [inline templates/rust/common.tera:692]: I think this should be removed `and at the actual offset`

File: templates/rust/common.tera [hunk 7]
```diff
@@ -668,12 +877,21 @@ where
             marker: PhantomData,
         }
     }
+
+    /// Get mask for bitfield, mask is unshifted and at the actual offset
```
Comment (pellico) [inline templates/rust/common.tera:881]: I think this should be removed `and at the actual offset`. At least it seems confusing

**Review by pellico — CHANGES_REQUESTED** [review]

File: templates/rust/common.tera (hunk 1 above)
Comment (andreasWallnerIFX) [reply templates/rust/common.tera:191]: Not sure I follow: this is for the RegValue - there is no masking here, and there are no bitfields. `set_raw` sets the value independent of possible bitfields in the register. Did you maybe mix it up with `RegisterField`?

File: templates/rust/common.tera (hunk 6 above)
Comment (andreasWallnerIFX) [reply templates/rust/common.tera:692]: I put it there because I found "mask is unshifted" to be unclear. Does "unshifted" mean that it's at the LSB and not shifted into the actual position - or does "unshifted" mean that it's at the actual position and not shifted down. It depends on which assumption you are starting with.
Reworded now to "the mask is shifted to the position of the bitfield" - that doesn't repeat itself and should be clear?

File: templates/rust/common.tera (hunk 7 above)
Comment (andreasWallnerIFX) [reply templates/rust/common.tera:881]: same as above^^

**Review by pellico — APPROVED** [review]
